examples/window-app/nrfconnect/main/WindowCovering.cpp

```diff
@@ -0,0 +1,340 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "WindowCovering.h"
+#include "AppConfig.h"
+#include "PWMDevice.h"
+
+#include <dk_buttons_and_leds.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/util/af.h>
+#include <logging/log.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <zephyr.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::WindowCovering;
+
+static k_timer sLiftTimer;
+static k_timer sTiltTimer;
+static constexpr uint32_t sMoveTimeoutMs{ 200 };
+
+static constexpr uint32_t FromOneRangeToAnother(uint32_t aInMin, uint32_t aInMax, uint32_t aOutMin, uint32_t aOutMax,
```

Comment:
What is the goal of this function 
There is a convert fonction available in the SDK with Window covering cluster please use it

Reply:
A some point I need to map the window position to the 0-255 range which is required by PWM driving LED indicators. If  there already is such a function in the SDK, could you please point it out?

Reply:
Those are static but you might want to expose them

Reply:
Done, thanks for the hint.